kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/BatchesResource.scala

```diff
@@ -525,22 +538,105 @@ private[v1] class BatchesResource extends ApiRequestContext with Logging {
       }
     }
   }
+
+  private def handleUploadingFiles(
+      batchId: String,
+      request: BatchRequest,
+      resourceFileInputStream: InputStream,
+      resourceFileName: String,
+      formDataMultiPartOpt: Option[FormDataMultiPart]): Option[JPath] = {
+    val uploadFileFolderPath = batchResourceUploadFolderPath(batchId)
+    handleUploadingResourceFile(
+      request,
+      resourceFileInputStream,
+      resourceFileName,
+      uploadFileFolderPath)
+    handleUploadingExtraResourcesFiles(request, formDataMultiPartOpt, uploadFileFolderPath)
+    Some(uploadFileFolderPath)
+  }
+
+  private def handleUploadingResourceFile(
+      request: BatchRequest,
+      inputStream: InputStream,
+      fileName: String,
+      uploadFileFolderPath: JPath): Unit = {
+    try {
+      val tempFile = Utils.writeToTempFile(inputStream, uploadFileFolderPath, fileName)
+      request.setResource(tempFile.getPath)
+    } catch {
+      case e: Exception =>
+        throw new RuntimeException(
+          s"Failed handling uploaded resource file $fileName: ${e.getMessage}",
+          e)
+    }
+  }
+
+  private def handleUploadingExtraResourcesFiles(
+      request: BatchRequest,
+      formDataMultiPartOpt: Option[FormDataMultiPart],
+      uploadFileFolderPath: JPath): Unit = {
+    val extraResourceMap = request.getExtraResourcesMap.asScala
+    if (extraResourceMap.nonEmpty) {
+      val fileNameSeperator = ","
+      val formDataMultiPart = formDataMultiPartOpt.get
+      val transformedExtraResourcesMap = extraResourceMap
+        .mapValues(confValue =>
+          confValue.split(fileNameSeperator).filter(StringUtils.isNotBlank(_)))
+        .filter { case (confKey, fileNames) =>
+          fileNames.nonEmpty && StringUtils.isNotBlank(confKey)
+        }.mapValues { fileNames =>
+          fileNames.map(fileName =>
+            Option(formDataMultiPart.getField(fileName))
+              .getOrElse(throw new RuntimeException(s"File part for file $fileName not found")))
+        }.map {
+          case (confKey, fileParts) =>
+            val tempFilePaths = fileParts.map(filePart => {
```

Comment:
nit: for multi lines lambda, `{` is always preferred than `(`
```suggestion
            val tempFilePaths = fileParts.map { filePart =>
```

Reply:
Thanks for the hint. Resoved.